File: Practice Solutions/9-19-19/bearSums.py
for tc in range(int(input())):
    valSet = set()
    S, E = [int(n) for n in input().split()]
    for i in [int(n) for n in input().split()]:
        if S - i in valSet:
            print(' '.join(str(val) for val in sorted([S-i,i])))
            break
        valSet.add(i)
    else:
        print("!OK")
        
# https://csacademy.com/ieeextreme-practice/task/bear-sums/statement/

File: Practice Solutions/9-19-19/bearSums.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int tc, sum, n, num;
    cin >> tc;
    while(tc--){
        set<int> valSet;
        bool found = false;
        cin >> sum >> n;
        while(n--){
            cin >> num;
            if(!found && valSet.find(sum - num) != valSet.end()){
                found = true;
                if(num > sum - num)
                    cout << sum - num << ' ' << num << endl;
                else 
                    cout << num << ' ' << sum - num << endl;
            }
            valSet.insert(num);

        }
        if(!found)
            cout << "!OK" << endl;
    }
}

// https://csacademy.com/ieeextreme-practice/task/bear-sums/statement/